api/rest/client.go

```diff
@@ -129,6 +129,10 @@ func (c *Client) DoRequest(req *http.Request, resp interface{}) (int, error) {
 	return httpResp.StatusCode, nil
 }
 
+func (c *Client) Do(req *http.Request) (*http.Response, error) {
```

Comment:
I know it's not your fault that `rest.Client` is already a leaky abstraction, but it seems that wherever you call `Do`, you're doing things very similarly to `DoRequest`. 
Is there a way to re-use, maybe change, the existing `DoRequest` method, rather than creating a new, leakier method?